### PR TITLE
Voter sees percentage #17

### PR DESCRIPTION
**Pull request by rickbacci**
@joshcheek @selfup @jbrr @rrgayhart 

My goal here is to iterate over the polls keys (split them into the admin/voter strings) and find the one that matches either an admin key, or a voter key.
If i find a match I will build the id (adminString + voterString) and use this to get the poll. I will also set an attribute to know which type of link is being looked at. Then I can show/hide things in the view based on
who it is.

After I have the poll I can pass it to the template.  Thoughts?

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -86,4 +80,53 @@ var server = http.createServer(app).listen(port, function () {
...
+    return redis;
+  } else {
+    var redis = require("redis").createClient();
+    return redis;
+  }
+}
+
+function logRedisKeys(redis) {
+  redis.hkeys('polls', function (err, keys) {
+    keys.forEach(function (key) {
```
Comment (selfup) [inline server.js]: Looks like you got the iteration down

File: server.js [hunk 2]
```diff
@@ -86,4 +80,53 @@ var server = http.createServer(app).listen(port, function () {
...
+}
+
+function storePoll(poll_id, poll, redis) {
+  redis.hmset(poll_id, poll)
+}
+
+
+// TODO move Poll to it's own file
+
+function Poll(poll_id, adminString, voterString, request) {
```
Comment (selfup) [inline server.js]: Making a class? Interesting!

You going to make methods soon?

File: server.js (hunk 1 above)
Comment (rickbacci) [inline server.js]: Holy crap, I forgot about that...lol. Thanks. @selfup

File: server.js (hunk 2 above)
Comment (rickbacci) [inline server.js]: I may break it out into prototype methods, i'm thinking the urls should be extracted for sure
@selfup Do i have to put your name in every post to notify you?

Comment (selfup) [inline server.js]: No you are good to go!

I am quite busy right now lol

**Comment by JoshCheek** [comment]
Nice job, Ricky! ^_^ Keep kicking ass!

**Comment by rickbacci** [comment]
@joshcheek @selfup @jbrr @rrgayhart

Not having luck with the way i envisioned this. Had a ton of problems with getting the data out of redis, so i scraped it for now. I'm also having problems with the render in show. It works fine on the redirect, but when you click a url the data disappears.

**Comment by selfup** [comment]
Do you know why the data disappears?

Have you messed around with updating and seeing if the data comes through?

**Comment by rickbacci** [comment]
I need to continue on this track now and implement the socket connection for voting

**Comment by rickbacci** [comment]
@joshcheek @selfup @jbrr @rrgayhart

**Comment by jbrr** [comment]
This is a pretty huge PR, I think it would take me a while to go through all of it. The poll class looks good though, I like the OO approach! Tests are passing? Things are working as you'd expect? What's your next step?

**Comment by selfup** [comment]
Agreed with Jeff. But it is merged. Guess the tests passed! What is the next step?

**Comment by rickbacci** [comment]
yeah but there is something wierd going on with the data. I need to figure
out a way to clear it after the tests.

**Comment by jbrr** [comment]
I'm not sure how Nightwatch works, but I have a beforeEach function in my tests that reset the poll object to either a blank object or a fixture, depending on the case. Would something like that be possible?

**Comment by rickbacci** [comment]
@jbrr yeah i might be able to do that but these are integration tests so i
was clicking thru

**Comment by jbrr** [comment]
Yeah, true. I know in Capybara there's a way to reset the session between tests, maybe there's something like that?

**Pull request merged**